### PR TITLE
28086 - "--downlevelIteration errors should mention using later targets"

### LINKED ISSUE #28086: --downlevelIteration errors should mention using later targets
```js
// @ts-check
class Foo {
  constructor(...args) {}
}
function makeFoo() {
  return new Foo(...arguments)
}
```

**Actual**

> Type 'IArguments' is not an array type. Use compiler option '--downlevelIteration' to allow iterating of iterators.

**Expected**

> Type 'IArguments' is not an array type. Either use the '--downlevelIteration' compiler option to allow iterating on iterators, or set the '--target' option to 'es2015' or above.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #28086 

@DanielRosenwasser

### CODE REVIEW

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser could you review this PR?

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser could you review this PR?

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser could you review this PR?

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -2048,7 +2048,7 @@
         "category": "Error",
         "code": 2567
     },
-    "Type '{0}' is not an array type. Use compiler option '--downlevelIteration' to allow iterating of iterators.": {
+    "Type '{0}' is not an array type. Either use the '--downlevelIteration' compiler option to allow iterating on iterators, or set the '--target' option to 'es2015' or above.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: What about the below error message?

Comment (a-tarasyuk) [reply src/compiler/diagnosticMessages.json]: Ok, will change. Thanks ;)

Comment (a-tarasyuk) [reply src/compiler/diagnosticMessages.json]: ?

Comment (a-tarasyuk) [reply src/compiler/diagnosticMessages.json]: @DanielRosenwasser ?

Comment (a-tarasyuk) [reply src/compiler/diagnosticMessages.json]: @DanielRosenwasser could you add to comment message which needs to use? Or point me out where you posted it?

Comment (a-tarasyuk) [reply src/compiler/diagnosticMessages.json]: @DanielRosenwasser ?

Comment (DanielRosenwasser) [reply src/compiler/diagnosticMessages.json]: Sorry, for context, there's an error directly below:

```
Type '{0}' is not an array type or a string type. Use compiler option '--downlevelIteration' to allow iterating of iterators.
```

Comment (a-tarasyuk) [reply src/compiler/diagnosticMessages.json]: Ok, will change. Thanks

Comment (a-tarasyuk) [reply src/compiler/diagnosticMessages.json]: @DanielRosenwasser fixed :+1:

**Comment by DanielRosenwasser** [comment]
I'm so confused - GitHub isn't showing a diff for the `diagnosticMessages.json` file at all...

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser I think that because was applied error 2569 and error 2568 was not changed. Or 2568 should be removed?

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser I think that because was applied error 2569 and error 2568 was not changed. Or 2568 should be removed?

**Comment by a-tarasyuk** [comment]
@sheetalkamat For some reasons, CI-Node-6 is failed. Tests passed, however, there are some other issues, which seem don't relate to PR.

**Pull request merged**